src/shell-interface.h

```diff
@@ -87,17 +87,21 @@ struct ShellExternalInterface : ModuleInstance::ExternalInterface {
   } memory;
 
   std::unordered_map<Name, std::vector<Literal>> tables;
+  std::map<Name, ModuleInstance*> linkedInstances;
 
   ShellExternalInterface() : memory() {}
   virtual ~ShellExternalInterface() = default;
 
-  void init(Module& wasm, ModuleInstance& instance) override {
+  void init(Module& wasm,
+            ModuleInstance& instance,
+            std::map<Name, ModuleInstance*> linkedInstances) override {
+    linkedInstances.swap(linkedInstances);
```

Comment:
probably the linked instances should arrive as `std::shared_ptr`s. That way they can be passed around safely. (And we don't care about cyclic leaks.)

Reply:
Agreed. It uses `std::shared_ptr`s now.